test/langtools/tools/javac/processing/model/util/types/TestAnnotationStripping.java

```diff
@@ -0,0 +1,276 @@
+                 * a method. The general methodology is to verify that
+                 * types that can appear as return types when
+                 * annotated with type annotations appear as specified
+                 * as the result of type operations or when new types
+                 * are constructed.
+                 */
+
+                TypeMirror returnType = m.getReturnType();
+
+                System.err.println("Checking " + returnType);
+
+                testVacuous(returnType);
+                checkDeepEmptyAnnotations(typeUtils.stripAnnotations(returnType));
+
+                checkExpectedTypeAnnotations(returnType, expectedAnnotation);
+
+                // Note: the result of Types.asElement is *not*
+                // checked for its annotations since the return value
+                // is an Element and not a TypeMirror.
+
+                System.err.print("\tcapture()");
+                checkDeepEmptyAnnotations(typeUtils.capture(returnType));
+
+                System.err.print("\terasure()");
+                checkDeepEmptyAnnotations(typeUtils.erasure(returnType));
+
+                System.err.print("\tgetArrayType()");
+                ArrayType arrayType = typeUtils.getArrayType(returnType);
+                checkEmptyAnnotations(arrayType);
+                /*
+                 * "Annotations on the component type are preserved."
+                 */
+                checkEqualTypeAndAnnotations(returnType, arrayType.getComponentType());
+
+                if (!returnType.getKind().isPrimitive()) {
+                    /*
+                     * "Annotations on the bounds are preserved."
+                     */
+                    WildcardType wcType;
+                    checkEmptyAnnotations(wcType = typeUtils.getWildcardType(returnType, null));
+                    checkEqualTypeAndAnnotations(returnType, wcType.getExtendsBound());
+
+                    checkEmptyAnnotations(wcType = typeUtils.getWildcardType(null,       returnType));
+                    checkEqualTypeAndAnnotations(returnType, wcType.getSuperBound());
+                }
+
+                System.out.println(returnType.getAnnotation(TestTypeAnnotation.class));
+                System.out.println(returnType.getAnnotationsByType(TestTypeAnnotation.class).length);
+                TestTypeAnnotation ta = requireNonNull(returnType.getAnnotation(TestTypeAnnotation.class),
+                                                       returnType.toString());
+
+                System.err.println();
+                System.err.println();
+            }
+
+            if (failures > 0)
+                throw new RuntimeException(failures + " failures occured.");
+        }
+        return true;
+    }
+
+    void testVacuous(TypeMirror tm ) {
+        try {
+            var result = vacuousTypes.stripAnnotations(tm);
+            messager.printError("Unexpected non-exceptional result returned" +  result);
```

Comment:
maybe need a space at the end of the string, to separate "returned" from `result`

Reply:
Good catch.